zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/DefaultTransactionContext.java

```diff
@@ -33,8 +34,8 @@ public void runInTransaction(final TransactionOperation operations) {
       } else {
         runInNewTransaction(operations);
       }
-    } catch (final RecoverableException recoverableException) {
-      throw recoverableException;
+    } catch (final RecoverableException | UnrecoverableException reportableException) {
```

Comment:
@npepinpe when we talked about it, you suggested to not handle `RecoverableException` and `UnrecoverableException` differently and instead just rethrow all exceptions that are not `RocksDbException`. IMO it'd be okay to keep it as is. The last catch is just used for all unexpected exceptions, which is also reflected in the exception message.

Reply:
:+1: Like I said, it just looks funky, mostly due to the names (i.e. if you see a recoverable, or an unrecoverable, just do the same :smile:)